ethereum-rpc/src/constructors/express/routes/api/v1/tokens.js

```diff
@@ -1,43 +1,168 @@
-const axios = require('axios')
-const router = require('express').Router()
+const axios = require("axios");
+const router = require("express").Router();
 
+const prefixContractAddress = (address) =>
+  address.startsWith("0x") ? address : `0x${address}`;
 
-module.exports = ({ models, ethereum: { web3, buildContract, compiler }, ...context }) => {
-
-  router.post('/', async (req, res, next) => {
+module.exports = ({
+  models,
+  ethereum: { web3, buildContract, compiler, abi },
+  ...context
+}) => {
+  router.post("/", async (req, res, next) => {
     try {
       const {
-        network = 'mainnet',
-        type = 'UNISWAP',
+        network = "mainnet",
+        type = "UNISWAP",
         amount = 0,
-        sender = '',
-        recipient = '',
-        tokenContractAddress = '',
-        privateKey = '',
+        sender = "",
+        recipient = "",
+        tokenContractAddress = "",
+        privateKey = "",
         sendOptions = {
           gasLimit: web3.utils.toHex(100000),
           gasPrice: web3.utils.toHex(50e9),
         },
         token = {},
         ...body
-      } = req.body
+      } = req.body;
 
       const result = await compiler.deployContract({
         token,
         type,
         sender,
         privateKey,
         sendOptions,
-      })
+      });
+
+      console.log("result", result);
+
+      return res.status(200).json(result.deployTransaction);
+    } catch (e) {
+      console.error(e);
+      return res.status(500).json({ errors: [e] });
+    }
+  });
+
+  router.get("/:address/balance", async (req, res, next) => {
+    try {
+      const { address = "*" } = req.params;
+      let { tokenAddress = "", type = "erc20" } = { ...req.query };
+      const tokenContractAddress = prefixContractAddress(
+        tokenAddress.toString("hex")
+      );
+      if (!web3.utils.isAddress(tokenContractAddress)) {
+        return res.status(500).json({ error: "invalid contract address" });
+      }
+      if (!web3.utils.isAddress(address)) {
+        return res.status(500).json({ error: "invalid address" });
+      }
+      let tokenType = type.toString();
+      if (!(tokenType == "erc20" || tokenType == "erc721")) {
+        return res.status(500).json({ error: "invalid token type" });
+      }
```

Comment:
With parameters with these, it may be ideal to force them to either uppercase or lowercase before moving forward with comparisons and other logic. And when we compare them, it would be good to use ENUMs / string constants

Reply:
ok, the code has been changed to the following
```
const tokenType = type.toString().toUpperCase();
      if (!(tokenType == "ERC20" || tokenType == "ERC721")) {
        return res.status(500).json({ error: "invalid token type" });
      }
```